### PR TITLE
Add BFM card references for MarkdownDef (CS-10570)

### PR DESCRIPTION
**Pull request by lukemelia**
## Summary
- Add Boxel Flavored Markdown card reference support: `:card[URL]` (inline, atom format) and `::card[URL]` (block, embedded format)
- New shared parsing utility (`bfm-card-references.ts`) with keyword-generic design for future `:file` support (CS-10583)
- MarkdownDef gains `cardReferenceUrls` (computed) and `linkedCards` (query-backed `linksToMany`) fields
- MarkdownTemplate renders linked cards into BFM placeholder elements via modifier + `{{in-element}}`

## Test plan
- [x] 22 unit tests for extraction, marked extensions, HTML generation, sanitization, and edge cases (code blocks, dedup, special chars)
- [x] Acceptance tests for inline/block card rendering and fallback text (require full test infrastructure)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

image

### CODE REVIEW

**Review by burieberry — COMMENTED** [review]
This is cool. However when there's `.json` in the URL, the card embed doesn't work. We should handle that, too.

**Comment by lukemelia** [comment]
> This is cool. However when there's .json in the URL, the card embed doesn't work. We should handle that, too.

Added.

File: packages/runtime-common/realm-index-query-engine.ts [hunk 1]
```diff
@@ -442,6 +442,20 @@ export class RealmIndexQueryEngine {
     return await this.#indexQueryEngine.getFile(url, opts);
   }
 
+  async loadLinksForResource(
```
Comment (habdelra) [inline packages/runtime-common/realm-index-query-engine.ts]: hmmm, this is making my spidey senses tingle, tell me more about how this is used. I want to make sure that it plays nice with how the store is loading links. we always lazily load links based on templates pulling on the links. i want to make sure this doesn't cause something to violate that.

Comment (lukemelia) [reply packages/runtime-common/realm-index-query-engine.ts]: This brings to file-def JSON-API the same thing we do for card-def JSON-API, which is to sideload any referenced linksTo/linksToMany cards in the `included` node

**Review by burieberry — APPROVED** [review]

File: packages/runtime-common/realm-index-query-engine.ts (hunk 1 above)
Comment (habdelra) [reply packages/runtime-common/realm-index-query-engine.ts]: can you point to a callsite where this is used? i'm actually having trouble finding it

Comment (lukemelia) [reply packages/runtime-common/realm-index-query-engine.ts]: Good catch! It got lost. Let me get it back in there.

**Review by habdelra — APPROVED** [review]
